lib/src/main/java/graphql/nadel/validation/NadelEnumValidation.kt

```diff
@@ -6,35 +6,34 @@ import graphql.schema.GraphQLEnumType
 import graphql.schema.GraphQLEnumValueDefinition
 
 internal class NadelEnumValidation {
+    context(NadelValidationContext)
     fun validate(
         schemaElement: NadelServiceSchemaElement,
-    ): List<NadelSchemaValidationError> {
+    ): NadelSchemaValidationResult {
```

Comment:
Commenting here about `NadelSchemaValidationResult` as this is the first reference in the diff.

`NadelSchemaValidationResult` is the root class for all schema validation results.
`NadelSchemaValidationError` extends from `NadelSchemaValidationResult`
We also have other `NadelSchemaValidationResult` implementations that actually return data
e.g. `NadelValidatedFieldResult` , `NadelValidatedTypeResult` and `NadelReachableServiceTypesResult` etc.

This is how we return more than just errors.

You'll also notice that we removed the `List` from here. That's because one implementation of `NadelReachableServiceTypesResult` is a wrapper around `List`.
This is done to make passing around the result easier.